Structure_Engine/Query/SolidVolume.cs

```diff
@@ -106,6 +104,34 @@ public static double SolidVolume(this PileFoundation pileFoundation)
             return pileFoundation.IsNull() ? 0 : pileFoundation.PileCap.SolidVolume() + pileFoundation.Piles.Select(x => x.SolidVolume()).Sum();
         }
 
+        /***************************************************/
+
+        [Description("Returns a Stem's solid volume based on the area of the element and its top and bottom thickness.")]
+        [Input("stem", "The Stem to get the volume from.")]
+        [Output("volume", "The Stem solid material volume.", typeof(Volume))]
+        public static double SolidVolume(this Stem stem)
+        {
+            return stem.IsNull() ? 0 : stem.Outline.Area() * (stem.ThicknessBottom + stem.ThicknessTop) / 2;
+        }
+
+        /***************************************************/
+
+        [Description("Returns a RetainingWall's solid volume based on the Stem and Footing volumes.")]
+        [Input("retainingWall", "The RetainingWall to get the volume from.")]
+        [Output("volume", "The RetainingWall solid material volume.", typeof(Volume))]
+        public static double SolidVolume(this RetainingWall retainingWall)
+        {
+            if (retainingWall.Footing.TopOutline.ControlPoints().OrderBy(p => p.Z).First().Z < retainingWall.Stem.Outline.ControlPoints().OrderBy(p => p.Z).First().Z)
```

Comment:
```suggestion
        [Description("Returns a RetainingWall's solid volume by summing the volume of the Stem and Footing volumes.")]
        [Input("retainingWall", "The RetainingWall to get the volume from.")]
        [Output("volume", "The RetainingWall solid material volume.", typeof(Volume))]
        public static double SolidVolume(this RetainingWall retainingWall)
        {
            if (retainingWall.Footing.TopOutline.ControlPoints().OrderBy(p => p.Z).First().Z < retainingWall.Stem.Outline.ControlPoints().OrderBy(p => p.Z).First().Z)
```
We need to add tolerances for this if we are going to check this. I am of the opinion that we keep the method simple and adjust the descriptions appropriately. 

We can also have an `IsValid()` method in the future to check if the `RetainingWall` overlaps etc.

Reply:
Added an IsValid method and implemented it here.